### PR TITLE
Dev Portal UI & Responsive Changes

### PR DESCRIPTION
**Pull request by kars0nn**
# In this, I changed and added quite a few things to make the site look better.

- Changed the way the site looks a little bit.
- Made the site responsive for mobile devices.
- There is now a drop-down navigation menu similar to the one on 7tv.app, only for small screens.
- Added screen size breakpoints to the scss files. This allows for easier responsive implementation. 
- The route link panel now collapses and expands with the click of a button on smaller screens. 
- Content in the docs route no longer overflows outside of screen. 
- Added a feature where users can click on the path of an API route and copy it to their clipboard
- Put the method for API routes inside a colored container, this makes it easier to read.
- Added more meta tags for SEO purposes.
- Changed swag.Operation to swag.Spec so we can display parameter info without typescript yelling at us

### CODE REVIEW

File: portal/src/components/Nav.vue [hunk 1]
```diff
@@ -209,44 +271,52 @@ nav {
...
 		}
 	}
+
 	.toggle-collapse {
 		margin-left: auto;
 		display: block;
 	}
 }
-</style>
+</style>
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:322]: nit: newline

File: portal/src/router/router.ts [hunk 2]
```diff
@@ -3,18 +3,21 @@ import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
...
+	},
 ] as RouteRecordRaw[];
 
 const router = createRouter({
 	history: createWebHistory(import.meta.env.BASE_URL),
 	routes,
 });
 
-export default router;
+export default router;
```
Comment (TroyKomodo) [inline portal/src/router/router.ts:23]: nit: newline

File: portal/src/components/Nav.vue [hunk 3]
```diff
@@ -28,15 +28,29 @@
...
+		</div>
+	</div>
 </template>
 
 <script setup lang="ts">
-import { ref } from "vue";
+import { ref, watch } from "vue";
 import Logo from "@svg/Logo.vue";
+import { useRoute } from "vue-router";
+import { useClipboard, usePermission } from '@vueuse/core'
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:46]: Where is this used?

**Comment by TroyKomodo** [comment]
Could you please run the formatter & linter to catch unused imports and reformat the code to be our standard.

**Comment by kars0nn** [comment]
Will do

File: portal/index.html [hunk 4]
```diff
@@ -4,6 +4,11 @@
 		<meta charset="UTF-8" />
 		<link rel="icon" type="image/svg+xml" href="/ico.svg" />
 		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+		<meta name="title" content="7tv API" />
```
Comment (TroyKomodo) [inline portal/index.html:7]: Could you refer to 7tv as `7TV`

File: portal/src/components/Nav.vue [hunk 5]
```diff
@@ -187,15 +222,39 @@ nav {
...
 		border-color: #303030;
 	}
+
 	&:active {
 		background-color: #424242;
 	}
 }
 
 @media screen and (max-width: 1000px) {
+	// made my own class style bcause holy cheese and crackers this code before was so unreadable
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:236]: Could you remove these comments. Or reword them so they are actually useful.

File: portal/src/views/Docs/Docs.vue [hunk 6]
```diff
@@ -121,47 +178,183 @@ interface RouteDef {
...
+// Copy to clipboard stuff - lets users copy the request url to their clipboard with ease
+const { isSupported, copy } = useClipboard();
 </script>
 
 <style scoped lang="scss">
+// TODO clean up these style defs in the future
+
 @import "@style/themes.scss";
 
+body {
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:193]: Does this style do anything since its a scoped style?

File: portal/src/views/Docs/Docs.vue [hunk 7]
```diff
@@ -121,47 +178,183 @@ interface RouteDef {
...
+	width: 100%;
+	max-width: 500px;
+}
+
+pre {
+	overflow-x: auto;
+	/* Use horizontal scroller if needed */
+	white-space: pre-wrap;
+	/* css-3 */
+	white-space: -moz-pre-wrap !important;
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:235]: Why did we use !important here?

Comment (TroyKomodo) [reply portal/src/views/Docs/Docs.vue:235]: Also, with this CSS does the preprocessor not have styles for all browsers?

File: portal/src/components/Nav.vue [hunk 8]
```diff
@@ -163,18 +180,36 @@ nav {
 	height: 100%;
 	border-bottom: 0.1em solid transparent;
 	font-size: 0.85em;
-	&.router-link-active {
-		border-color: currentColor;
+	transition: all 0.2s ease-in-out;
+
+	@include breakpoint(md, max) {
+		&.router-link-active {
+			background-color: rgba(202, 0, 0, 0.707);
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:187]: Could we use hex numbers?

File: portal/src/components/Nav.vue [hunk 9]
```diff
@@ -187,15 +222,39 @@ nav {
...
 @media screen and (max-width: 1000px) {
+	// made my own class style bcause holy cheese and crackers this code before was so unreadable
+	// still using the nav-link class at the moment. This will be cleaned up.
+	.realNav {
+		@include breakpoint(lg, min) {
+			display: none;
+		}
+
+		transition: all 0.5s ease-in-out;
+		font-size: 24px;
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:244]: Could we use relative units?

File: portal/src/components/Nav.vue [hunk 10]
```diff
@@ -187,15 +222,39 @@ nav {
...
+		display: block;
+		margin-top: 4.5rem;
+		z-index: 150;
+		position: fixed;
+		width: 100vw;
+		box-shadow: black 8px 0 15rem;
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:250]: Could we use relative units?

File: portal/src/components/Nav.vue [hunk 11]
```diff
@@ -187,15 +222,39 @@ nav {
...
+		backdrop-filter: blur(15px);
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:251]: Could we use relative units?

File: portal/src/views/Docs/Docs.vue [hunk 12]
```diff
@@ -83,19 +139,20 @@ const routes = computed(() => {
...
+				params: params as swag.Spec,
 				method,
 				path: key,
 			});
 		}
 	}
 
-	return result;
+	return {
+		result, // i'm too lazy to take this out of the object...
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:154]: What?

File: portal/src/views/Docs/Docs.vue [hunk 13]
```diff
@@ -121,47 +178,183 @@ interface RouteDef {
...
+	align-items: center;
+	justify-content: center;
+	flex-direction: column;
+	padding: 16px;
+}
+
+.accordions {
+	width: 100%;
+	max-width: 500px;
+}
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:228]: Could we use relative units?

File: portal/src/views/Docs/Docs.vue [hunk 14]
```diff
@@ -121,47 +178,183 @@ interface RouteDef {
...
-				color: themed("warning");
+				background-color: #d7392e;
 			}
 		}
 
 		.route-item-description {
-			margin: 0.5em;
+			border-radius: 0 0 5px 5px;
+			padding: 10px;
+			color: white;
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:357]: Could we use relative units?

File: portal/src/views/Docs/Docs.vue [hunk 15]
```diff
@@ -209,4 +446,24 @@ main.docs {
 		}
 	}
 }
+
+.toggle-collapse {
+	display: none;
+	background-color: #373737;
+	font: inherit;
+	color: inherit;
+	margin-left: 5px;
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:455]: Could we use relative units?

**Review by TroyKomodo — CHANGES_REQUESTED** [review]
Try using relative units `rem` or `em` over hard coded `px` values where we can.

File: portal/src/views/Docs/Docs.vue (hunk 6 above)
Comment (kars0nn) [reply portal/src/views/Docs/Docs.vue:193]: used this for debugging _something_ but it's not needed anymore, I took it out just now

File: portal/src/views/Docs/Docs.vue (hunk 7 above)
Comment (kars0nn) [reply portal/src/views/Docs/Docs.vue:235]: i took that from a stack-overflow answer to fix the <pre> overflowing off of the screen, but not sure why they put that extra stuff in, maybe best practice? i took it out

File: portal/src/views/Docs/Docs.vue [hunk 16]
```diff
@@ -55,6 +104,13 @@
 <script setup lang="ts">
```
Comment (AnatoleAM) [inline portal/src/views/Docs/Docs.vue:104]: This file should be split into multiple now

File: portal/src/views/Docs/Docs.vue [hunk 17]
```diff
@@ -10,39 +10,88 @@
 						<option v-for="v of versions">{{ v }}</option>
 					</select>
 				</span>
+				<button class="toggle-collapse" @click="toggleNav">Expand Links</button>
```
Comment (AnatoleAM) [inline portal/src/views/Docs/Docs.vue:13]: Make the sidebar its own component (`DocsSidebar.vue`)

File: portal/src/views/Docs/Docs.vue [hunk 18]
```diff
@@ -10,39 +10,88 @@
...
 				<div class="route-navigator">
-					<div class="route-navigator-item" v-for="route in routes" :key="route.id">
+					<div class="route-navigator-item" v-for="route in routes.result" :key="route.id">
 						<a class="route-navigator-link" @click="scrollTo($event, route.id)">{{ route.summary }}</a>
 					</div>
 				</div>
 			</div>
 		</div>
-
 		<div class="docs-content">
```
Comment (AnatoleAM) [inline portal/src/views/Docs/Docs.vue:32]: Make content its own component (`DocsContent.vue`)

File: portal/src/views/Docs/Docs.vue [hunk 19]
```diff
@@ -10,39 +10,88 @@
...
 			<div v-if="error" class="load-error">
 				<span>{{ error }}</span>
 			</div>
-
 			<div>
 				<div class="route-list">
 					<div class="route-list">
-						<div class="route-item" v-for="route of routes">
+						<div class="route-item" :method="route.method" v-for="route of routes.result">
```
Comment (AnatoleAM) [inline portal/src/views/Docs/Docs.vue:39]: Make route item its own component (`DocsRouteItem.vue`)

File: portal/src/views/Docs/Docs.vue (hunk 12 above)
Comment (kars0nn) [reply portal/src/views/Docs/Docs.vue:154]: AINTNOWAY

File: portal/src/views/Docs/Docs.vue [hunk 20]
```diff
@@ -10,39 +10,88 @@
...
+											string
+											<!-- change this soon, don't want this to be hardcoded in, but the object doesn't have a type apparently? -->
+											<br />
+											<small
+												><i>({{ f?.in ?? "no data input type" }})</i></small
+											>
+											<br />
+											<p style="font-weight: bold">{{ f?.description ?? "no description" }}</p>
+											<pre style="margin-top: 5px">{{
+												route.params.consumes ?? "no filetype data"
```
Comment (AnatoleAM) [inline portal/src/views/Docs/Docs.vue:80]: When there is no consume type, simply omit this part (use `v-if`)

**Review by AnatoleAM — CHANGES_REQUESTED** [review]
Visually pretty nice, just some tidying is needed

**Comment by kars0nn** [comment]
Updated the px to rem or em, changed rgb/rgba color values to hexadecimal color values, and created components for the DocsSidebar and DocsRouteItem

File: portal/src/components/Nav.vue [hunk 21]
```diff
@@ -187,15 +222,37 @@ nav {
...
 	&:hover {
 		border-color: #303030;
 	}
+
 	&:active {
 		background-color: #424242;
 	}
 }
 
 @media screen and (max-width: 1000px) {
```
Comment (TroyKomodo) [inline portal/src/components/Nav.vue:235]: Why have this breakpoint if we are going to use breakpoints under it?

File: portal/src/views/Docs/Docs.vue [hunk 22]
```diff
@@ -121,89 +83,93 @@ interface RouteDef {
 	description: string;
 	method: string;
 	path: string;
-	params: swag.Operation;
+	params: swag.Spec; // changed to swag.Spec because it fit our needs better than swag.Operation
```
Comment (TroyKomodo) [inline portal/src/views/Docs/Docs.vue:86]: Can you comment above this why swag spec fits out needs better than swag operation.
So essentially explain why you believe that swag.Spec is required here. or what it does?

**Comment by TroyKomodo** [comment]
Looks really good tho!

File: portal/src/components/Nav.vue (hunk 21 above)
Comment (kars0nn) [reply portal/src/components/Nav.vue:235]: Not sure, the @media tag was there before I started working on this, I will optimize it.

File: portal/src/App.vue [hunk 23]
```diff
@@ -24,6 +25,7 @@ useHead({
 		class: "theme-dark",
 	},
 });
+
```
Comment (TroyKomodo) [inline portal/src/App.vue:28]: Remove extra line. Were these added by the formatter?

File: portal/src/components/Docs/DocsSideBar.vue [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+	routes: any;
+}>();
+
+</script>
+
+<style scoped lang="scss">
+
+@import "@style/themes.scss";
+
+#sidebarmain {
```
Comment (TroyKomodo) [inline portal/src/components/Docs/DocsSideBar.vue:68]: Why are we using IDs on css variables, please use classes only.

File: portal/src/App.vue [hunk 25]
```diff
@@ -16,6 +16,7 @@
 </template>
 
 <script setup lang="ts">
+
```
[the remark this answers is not in the document]
Comment (TroyKomodo) [reply portal/src/App.vue:19]: I dont think we have extra lines at the beginning and end of script/style/template blocks.

**Comment by TroyKomodo** [comment]
Minor changes and then we are good I think!

File: portal/src/components/Docs/DocsSideBar.vue (hunk 24 above)
Comment (kars0nn) [reply portal/src/components/Docs/DocsSideBar.vue:68]: Fixed

File: portal/src/App.vue (hunk 23 above)
Comment (kars0nn) [reply portal/src/App.vue:28]: These were added by me, I removed them. Not sure why I did it.

**Review by AnatoleAM — APPROVED** [review]
LGTM

**Pull request merged**